src/main/kotlin/com/theapache64/dde/navigation/NavHostComponent.kt

```diff
@@ -49,23 +52,24 @@ class NavHostComponent(
      * Invoked when `GO` button clicked (InputScreen)
      */
     private fun onGoClicked(name: String) {
-        router.push(ScreenConfig.Greeting(name))
+        navigation.push(ScreenConfig.Greeting(name))
     }
 
     /**
      * Invoked when `GO BACK` button clicked (GreetingScreen)
      */
     private fun onGoBackClicked() {
-        router.pop()
+        navigation.pop()
     }
 
 
     /**
      * Renders screen as per request
      */
+    @OptIn(ExperimentalDecomposeApi::class)
     @Composable
     override fun render() {
-        Children(routerState = router.state) {
+        Children(stack){
```

Comment:
Maybe use one of the predefined [animations](https://arkivanov.github.io/Decompose/extensions/compose/#animations-experimental)?